first

sebastian-filip opened the pull request:
here it is, it's not perfect because i didn't spend time on adding enough elements on the data base and as a teacher you can't change grades, but on the other elements it works. I attached the sql script for mysql workbench if you want to check it out.

sebastian-filip committed:
commit

sebastian-filip committed:
2nd commit

Alex-Cosma commented:
Could you please put the mysql-connector library as well? Thanks!

sebastian-filip committed:
here it is

Alex-Cosma commented:
Numai `.jar`-ul, Sebi

filipsebastian commented:
nu pot sa pun pe git jar-ul. Ti-am pus pe wetransfer fisierul care contine si jar-ul.

Alex-Cosma commented:
Tu ai trecut la baze de date?

filipsebastian commented:
da... ii chiar asa rau ?

Alex-Cosma commented:
da

Alex-Cosma commented:
Cum sărăcia ai trecut la baze de date!?

filipsebastian commented:
ce am gresit ?

Alex-Cosma commented:
Deci, hai să încep cu partea de OOP și apoi o să trec și la baze de date:
- în primul rând tu creezi o conecțiune de genul
```
Connection myConn = DriverManager.getConnection(
					"jdbc:mysql://localhost:3306/DigitalSchool", "root", "");
```
de fiecare dată când faci un request la baza de date. Nu ar trebui să faci asta. Ar trebui să lași o conecțiune deschisă și să te folosești de ea. Cum faci asta? Simplu, ori o faci statică ori folosești singleton. 

- Ai două clase care sunt aproape 100% identice! Student login și Teacher Login.Eu ți-am zis cum să faci - o singură tabelă user care să aibă un field de `type`, care poate să fie ori student (0) ori teacher (1) sau mai știu eu ce mapare găsești. Așa nu ai avea nevoie de două clase pentru același lucru (login!)

- proiectul tău nu face aproape nimic decât să arate niște date. Ți-am zis lista de requirement-uri pe care o am în mail.


La baze de date:
- nu îmi vine să cred că ai trecut
- ai tabele pentru fiecare materie în parte. How is that a good thing? De ce nu te folosești de tabelul `Subject` pe care l-ai creat și să ai un tabel `Grades` sau `Marks` care să aibă așa: `subjectId`,`studentId`, `grade` și `date`? Astfel nu ai avea câte o tabelă pentru fiecare materie. That's not even first normal form.. :disappointed: 
- pattern-ul de la `teacherclassroom` e ok, cam așa ar trebui să fie

Alex-Cosma commented:
Să nu mai zic că nu respecți naming convention nici în db, nici în cod

Alex-Cosma commented:
E fain că ai folosit `JTable` acolo

filipsebastian commented:
inca nu stiu cum sa modific datele din baza de date. mai trebe sa caut pe net putin. si asta ar si trebui sa faca un catalog digital, sa arate note in marea parte. m-am gandit sa fac o tabela cu note, doar ca nu mi-a venit ideea aia. ii mult de schimbat acuma. Nu intelg de ce nu e bine cu 2 loginuri diferite ? adica nu te complici deloc, plus ar putea sa intre si un student si un profesor deodata pe cont.

Alex-Cosma commented:
Faptul că intră deodată poți rezolva prin a deschide de 2 ori programul.
Uite cât cod duplicat ai. 2 clase identice aproape, cum să fie bine?

filipsebastian commented:
si sa le fac o singura clasa ?

Alex-Cosma commented:
Da

Alex-Cosma commented:
Faptul că sunt multe de schimbat.. that is NOT an excuse!

sebastian-filip committed:
Am schimbat cateva chestii din ce mi-ai zis. Inca nu poti sa modifici note/absente ca si profesor, dar va urma. Ii ok in rest ? nu is sigur daca ii ok cum am facut conexiunea cu baza de date.

Alex-Cosma commented:
Șterge clasele pe care nu le folosești

Alex-Cosma commented:
E mult mai OK chestia cu conecțiunea, da.

Alex-Cosma commented:
Naming conventions!! Sunt foarte importante, te rog să le respecți și să-ți formatezi codul

filipsebastian commented:
In baza de date, Id-urile nu mai pot sa le redenumesc numa daca ii drop la table. Dar in cod o sa schimb tot ce pot.

sebastian-filip committed:
Am rezolvat, se pot schimba note daca esti logat ca si profesor. Nu am gasit metodata mai buna decat asta, si pe net tot ce am gasit ii ca trebe sa folosesti MouseListener. Singurul minus e ca daca vrei sa iti salveze, trebui sa scri in casuta din tabela, dupaia sa apesi cu mousu in alta casuta din tabel, pe urma sa selectezi casuta care ai modificato precedent din nou si sa apesi bunoul de SAVE.
Poate ca este ceva metoda mai buna dar eu nu am gasit... Si asa m-am chinuit destul de mult cu asta. Urmeaza sa modific chestii pe acolo, cu naming convension. Cam cati elevi sa adaug in baza de date, nu as adauga 30 de pe fiecre clasa, ii ok daca pun 2-3 in fiecare clasa ?

Alex-Cosma commented:
Da, poți pune <10 în fiecare clasă

Alex-Cosma commented:
Trebuie să muți logica care se ocupă cu statement-urile SQL în alte clase. Numește-le `Services` sau ceva, și fiecare service să se ocupe de o altă parte din code (note, login, etc.)

Dar încă nu ai rezolvat partea cu login-ul. Tot ai 2 pagini la fel aproape. Eu ți-am zis așa: un singur login și în funcție de ce tip e userul logat, să îl 'redirectezi' la pagina care trebuie. Asta înseamnă că ai o singură tabelă 'user' care are o coloană 'type' care dictează ce fel de user e (student / profesor)

filipsebastian commented:
Care is alea doua clase care seamana ? Clasele alea le-am sters miercuri cand mi-ai zis de ele prima data. Ii complet diferit acuma.

Alex-Cosma commented:
`StudenLogin` cu `TeacherLogin`

filipsebastian commented:
cred ca ai ceva versiune mai veche sau, nu s-or sters de pe git. dar acuma nu mai exista clasele alea este doar o singura clasa LogIn.

Alex-Cosma commented:
ouh, îmi pare rău atunci, dar ai pus noa variantă pe git? care e?

filipsebastian commented:
Acuma am verificat. Is aiurea copiate. Pun acum din nou,

sebastian-filip committed:
In pachetul de GUI trebe sa fie doar 4 clase nu 5.

Alex-Cosma commented:
Acum e ok

sebastian-filip committed:
Asta-i versiunea finala. Mege! Ce ti-am trimis data trecuta nu mergea pt fiecare profesor doar pt unii. Acum merge pentru orice profesor. Am adaugat si mai multi elevi in fiecare clasa in baza de date.
